test/e2e/framework/pod/wait.go

```diff
@@ -809,6 +809,18 @@ func WaitForPodContainerStarted(ctx context.Context, c clientset.Interface, name
 	})
 }
 
+// WaitForPodInitContainerStarted waits for the given Pod init container to start.
+func WaitForPodInitContainerStarted(ctx context.Context, c clientset.Interface, namespace, podName string, initContainerIndex int, timeout time.Duration) error {
```

Comment:
Not sure if we need it that specific. For example, `WaitForContainerRunning` allows to wait for any container to be running.

Reply:
> Not sure if we need it that specific. For example, `WaitForContainerRunning` allows to wait for any container to be running.

`WaitForContainerRunning` indeed meets the requirement. However, there is a check for `started` in `getPhase`. Additionally, since we already have `WaitForPodContainerStarted`, adding a `WaitForPodInitContainerStarted` seems reasonable.